ci-operator/step-registry/telco5g/origin/tests/telco5g-origin-tests-commands.sh

```diff
@@ -44,30 +41,236 @@ git clone -b "${CNF_BRANCH}" "${CNF_REPO}" cnf-features-deploy
+      ansible.builtin.shell:
+        cmd: kcli ssh 'root@${CLUSTER_NAME}-installer' 'chmod a+x /usr/bin/kubectl'
+
+    - name: Make kubeconfig available to non-root user on installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli ssh 'root@${CLUSTER_NAME}-installer' "sh -c '\\
+              cp /root/ocp/auth/kubeconfig /home/cloud-user/kubeconfig \\
+              && chown cloud-user /home/cloud-user/kubeconfig \\
+          '"
+
+    - name: Test kubeconfig for non-root user on installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli ssh 'cloud-user@${CLUSTER_NAME}-installer' "sh -c 'export KUBECONFIG=kubeconfig && oc get nodes'"
+
+    - name: Run script for openshift-tests binary on installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli ssh 'cloud-user@${CLUSTER_NAME}-installer' "/home/cloud-user/run-openshift-tests.sh"
+
+    - name: Compress results from openshift-tests binary on installer
+      ansible.builtin.shell:
+        cmd: |
+          kcli ssh 'cloud-user@${CLUSTER_NAME}-installer' \\
+            "tar -czf openshift-tests.tar.gz e2e.log junit/ openshift-tests.log || { rm -f openshift-tests.tar.gz; exit 1; }"
+
+    - name: Copy results from installer to bastion host
+      ansible.builtin.shell:
+        cmd: |
+          kcli scp 'cloud-user@${CLUSTER_NAME}-installer:/home/cloud-user/openshift-tests.tar.gz' \\
+            '/tmp/kcli_${CLUSTER_NAME}_openshift-tests.tar.gz'
+
+    - name: Copy results from bastion host to artifacts directory
+      ansible.builtin.fetch:
+        src: '/tmp/kcli_${CLUSTER_NAME}_openshift-tests.tar.gz'
+        dest: '${ARTIFACT_DIR}/openshift-tests.tar.gz'
+
+    always:
+    - name: Remove script to run openshift-tests binary from bastion host
+      ansible.builtin.file:
+        path: '/tmp/kcli_${CLUSTER_NAME}_run-openshift-tests.sh'
+        state: absent
+      ignore_errors: true
+
+    - name: Remove openshift-tests binary from bastion host
+      ansible.builtin.file:
+        path: '/tmp/kcli_${CLUSTER_NAME}_openshift-tests'
+        state: absent
+      ignore_errors: true
+
+    - name: Remove list of tests from bastion host
+      ansible.builtin.file:
+        path: '/tmp/kcli_${CLUSTER_NAME}_${TEST_FILE}'
+        state: absent
+      ignore_errors: true
+
+    - name: Remove openshift-tests results from bastion host
+      ansible.builtin.file:
+        path: '/tmp/kcli_${CLUSTER_NAME}_openshift-tests.tar.gz'
+        state: absent
+      ignore_errors: true
+EOF
+
+# Run OpenShift's conformance test suite
+ANSIBLE_STDOUT_CALLBACK=debug ansible-playbook -i "$SHARED_DIR/inventory" run-openshift-tests.yml -vv
+
+# TODO: Delete pull secrets?
+
+set +x # TODO: Drop
 
 set +e
-python3 -m venv ${SHARED_DIR}/myenv
-source ${SHARED_DIR}/myenv/bin/activate
-git clone https://github.com/openshift-kni/telco5gci ${SHARED_DIR}/telco5gci
-pip install -r ${SHARED_DIR}/telco5gci/requirements.txt
+python3 -m venv "${SHARED_DIR}/myenv"
```

Comment:
this whole section is probably not required, it's for parsing CNF tests results

Reply:
I was not sure whether you want to keep it here. Parts of it we probably want to reuse because we somehow want to postprocess test results so that you can generate your mails from it?